src/modules/hip/kernel/pre_emphasis_filter.hpp

```diff
@@ -0,0 +1,74 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ void pre_emphasis_filter_hip_compute(d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8, float4 *coeff_f4)
+{
+    dst_f8->f4[0] = src1_f8->f4[0]  - *coeff_f4 * src2_f8->f4[0];
+    dst_f8->f4[1] = src1_f8->f4[1]  - *coeff_f4 * src2_f8->f4[1];
+}
+
+__global__ void pre_emphasis_filter_tensor(float *srcPtr,
+                                           uint srcStride,
+                                           float *dstPtr,
+                                           uint dstStride,
+                                           Rpp32s *srcLengthTensor,
+                                           float *coeffTensor,
+                                           RpptAudioBorderType borderType)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= srcLengthTensor[id_z])
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStride) + id_x;
+    uint dstIdx = (id_z * dstStride) + id_x;
+
+    if(id_x == 0)
+    {
+        if(borderType == RpptAudioBorderType::ZERO)
+            dstPtr[dstIdx] = srcPtr[srcIdx];
+        else
+        {
+            float border = (borderType == RpptAudioBorderType::CLAMP) ? srcPtr[srcIdx] : srcPtr[srcIdx + 1];
+            dstPtr[dstIdx] = srcPtr[srcIdx] - coeffTensor[id_z] * border;
+        }
+    }
+
+    float4 coeff_f4 = (float4)coeffTensor[id_z];
+    d_float8 src1_f8, src2_f8, dst_f8;
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx + 1, &src1_f8);
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src2_f8);
+    pre_emphasis_filter_hip_compute(&src1_f8, &src2_f8, &dst_f8, &coeff_f4);
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx + 1, &dst_f8);
+}
+
+RppStatus hip_exec_pre_emphasis_filter_tensor(Rpp32f *srcPtr,
+                                              RpptDescPtr srcDescPtr,
+                                              Rpp32f *dstPtr,
+                                              RpptDescPtr dstDescPtr,
+                                              Rpp32s *srcLengthTensor,
+                                              RpptAudioBorderType borderType,
+                                              rpp::Handle& handle)
+{
+    int globalThreads_x = (dstDescPtr->strides.nStride + 7) >> 3;
+    int globalThreads_y = 1;
+    int globalThreads_z = handle.GetBatchSize();
```

Comment:
Use dstDescPtr->n instead of  handle.GetBatchSize() in all the new HIP kernels we are going to add

Reply:
Done